### DISCUSSION TITLE
Hive to Isar migration, Supabase integration, Riverpod v2 update

### DISCUSSION THREAD

**Discussion by helpisdev** [flutterdata/flutter_data#184] [General]
While browsing the web yesterday in search for offline-caching solutions I found this package. I was browsing the source today and I noticed that it uses Hive. I have no experience in Hive or Isar, but at both repositories Isar is suggested over Hive. Should this migration happen here as well? Would the project gain any benefit? Having no experience in none of the three packages I can't say for myself, but I would love to know what other people think. It seems interesting enough to me to start getting deeper into the source of all three packages and maybe make a PR for the migration.

Another thing I noticed while browsing is the usage of Riverpod v1 and some deprecated members of it in the new v2. The package exports riverpod, and this essentially restricts anyone from using riverpod v2 in their projects. Probably that's a sector that needs refactoring.

Finally, supabase_flutter uses hive for offline-caching as well, in a minimal ~100 loc file. I imagine that this package does the job more thoroughly, so I thought that an integration with supabase_flutter to replace their current implementation would benefit both projects, and would certainly attract more people to flutter_data as well.

All these are just things that crossed my mind while browsing and I shared them here without too much thought into them, but I'm very curious to see what other people think about them.

**Comment by frank06** [comment]
Thanks for your feedback.

Isar migration: I started [a migration branch](https://github.com/flutterdata/flutter_data/tree/isar) earlier this year. It's not simply a matter of swapping backends. I'm well aware of the effort and tradeoffs, development is halted simply because it's not in my highest priorities.

Supabase support using adapters should be very simple. Someone has to do it. In one of my projects we're simultaneously accessing three data sources just with FD adapters.

Riverpod: Yes, I plan to update that very soon.

**Comment by frank06** [comment]
FD 1.5.0 now supports the latest Riverpod
